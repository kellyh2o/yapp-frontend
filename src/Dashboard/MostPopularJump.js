import React from 'react';
import { connect } from "react-redux";
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Title from './Title';

const useStyles = makeStyles({
  depositContext: {
    flex: 2,
  },
});

function getMostPopularJump(demoLocation) {
  if (demoLocation && demoLocation.jumps) {
    const jumps = demoLocation.jumps;
    const orderedJumps = jumps.sort((x, y) => {
      return y.events.length - x.events.length;
    });
  
    return orderedJumps[0].name;
  } else {
    return "- - -";
  }
}

function getPopularJumpNumberOfHits(demoLocation) {
  if (demoLocation && demoLocation.jumps) {
    const jumps = demoLocation.jumps;
    const orderedJumps = jumps.sort((x, y) => {
      return y.events.length - x.events.length;
    });
  
    return orderedJumps[0].events.length;
  } else {
    return "- - -";
  }
}


const MostPopularJump = (props) => {
  const classes = useStyles();
  return (
    <React.Fragment>
      <Title>Today's Most Popular Jump</Title>
      <Typography component="p" variant="h4" style={{paddingTop: "20px"}}>
        {getMostPopularJump(props.demoLocation)}
      </Typography>
      <Typography color="textSecondary" className={classes.depositContext} style={{paddingTop: "20px", paddingLeft: "5px"}}>
        {getPopularJumpNumberOfHits(props.demoLocation)} Hits Today
      </Typography>
    </React.Fragment>
  );
}

const mapStateToProps = (state) => {
  return {
      demoLocation: state.demoLocation
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
      selectView: (view) => {dispatch({type: 'SELECT_VIEW', view: view})}
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(MostPopularJump);