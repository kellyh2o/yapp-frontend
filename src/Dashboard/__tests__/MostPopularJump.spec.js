import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import MostPopularJump from "../MostPopularJump";

describe('MostPopularJump', () => {
    it("should render", () => {
        const mockStore = configureMockStore();

        const initialState = {
            demoLocation: {
                name: "Hogwarts",
                jumps: [ 
                    {
                        name: "Jump1",
                        events: []
                    },
                    {
                        name: "Jump2",
                        events: []
                    },
                    {
                        name: "Jump3",
                        events: [
                            {
                                riderMass: 22,
                                riderSpeed: 12,
                            }
                        ]
                    }
                ]
            }
        };
        const store = mockStore(initialState);

        const { asFragment } = render(
            <Provider store={store}>
                <MostPopularJump />
            </Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    })
});

