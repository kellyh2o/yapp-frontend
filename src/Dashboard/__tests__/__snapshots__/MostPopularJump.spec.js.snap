// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostPopularJump should render 1`] = `
<DocumentFragment>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
  >
    Today's Most Popular Jump
  </h2>
  <p
    class="MuiTypography-root MuiTypography-h4"
    style="padding-top: 20px;"
  >
    Jump3
  </p>
  <p
    class="MuiTypography-root makeStyles-depositContext-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
    style="padding-top: 20px; padding-left: 5px;"
  >
    1 Hits Today
  </p>
</DocumentFragment>
`;
